Update LSP0 to latest specification (#492)
lukso-network/docs: docs/standards/smart-contracts/lsp0-erc725-account.md

---

## Behavior

All ownable functions such as `execute(..)`, `setData(..)`, `transferOwnership(..)`, and `renounceOwnership(..)` can be called by the owner and any address that is allowed by the owner conforming to the [LSP20-CallVerification](../universal-profile/lsp0-erc725account.md#lsp20---call-verification) behavior.

> If an address is calling the `execute(..)` function and is not the owner, the account contract will forward the call to the owner. The execution of the function will only continue if the owner returns a specific magic value, which indicates that the caller is allowed to execute the function. The magic value can also determine if there should be any post execution check on the owner. This same behavior applies to other ownable functions as well.

Reviewer: ```suggestion
If an address calls the `execute(..)` function and is not the owner, the account contract will forward the call to the owner. The execution of the function will only continue if the owner returns a specific magic value, indicating that the caller can execute the function. The magic value can also determine if there should be any post-execution check on the owner. This same behavior applies to other ownable functions as well.
```

> If an address is calling the `execute(..)` function and is not the owner, the account contract will forward the call to the owner. The execution of the function will only continue if the owner returns a specific magic value, which indicates that the caller is allowed to execute the function. The magic value can also determine if there should be any post execution check on the owner. This same behavior applies to other ownable functions as well.

Author: > indicating that the caller can execute the function

I still prefer "indicating that the caller can execute the function" because I see a difference between "is allowed" and "can". "Can" point to the ability to do it from the caller, and "is allowed to" point that the caller is allowed and could do the tx

> This allows the account to have a more dynamic and adaptable approach for managing function execution logic, utilizing the **LSP20-CallVerification** standard.

Reviewer: ```suggestion
The structure allows the account to have a more dynamic and adaptable approach for managing function execution logic, utilizing the **LSP20-CallVerification** standard.
```

2.Then return the return value received after this low level call.

> - If the data sent is **preprended with 4 zeros (0)**, then the call should be checked for an extension, if there is none the call should still pass.

Reviewer: ```suggestion
- If the data sent is **prepended with 4 zeros (0)**, then the call should be checked for an extension, if there is none the call should still pass.
```

>   For example, when sending value to the contract it can be associated with a message, such as `"Here is 1 Ether"`, after encoding the message as bytes: `0x486572652069732031204574686572`.

Reviewer: ```suggestion
  For example, sending value to the contract can be associated with a message, such as `"Here is 1 Ether" after encoding the message as bytes: `0x486572652069732031204574686572`.
```

>   For example, when sending value to the contract it can be associated with a message, such as `"Here is 1 Ether"`, after encoding the message as bytes: `0x486572652069732031204574686572`.

Reviewer 2: Suggest breaking it into a separate paragraph to make the _example_ section more visually noticeable:

```suggestion
_Example:_ 

When sending value to the contract with a message alongside the value transfer, such as `"Here is 1 Ether"`. The message encoded as bytes will be: `0x486572652069732031204574686572`.
```

>   In order to make the call to the contract pass, it should be prepended with `bytes4(0)`. A call to the **LSP0** with `0x00000000486572652069732031204574686572` as data **will pass** if there is no extension set. If there is an extension set for the `bytes4(0)` selector, the call will either pass or revert depending on the logic of the extension.

Reviewer: ```suggestion
  To make the call to the contract pass, it should be prepended with `bytes4(0)`. A call to the **LSP0** with `0x00000000486572652069732031204574686572` as data **will pass** if there is no extension set. If an extension is set for the `bytes4(0)` selector, the call will either pass or revert depending on the logic of the extension.
```

Requirements:

> - Can only be called by the current owner and any address allowed by the owner.

Reviewer: ```suggestion
- Can only be called by the current owner and any address the owner allows.
```

Leaves the contract without an owner. Once ownership of the contract is renounced, it won't be possible to call the functions restricted to the owner only.
:::

### batchCalls

```solidity
function batchCalls(bytes[] calldata data) public returns (bytes[] memory results)
```

> Allows a caller to batch different function calls in one call. Performs a delegatecall on self to call different functions while preserving the context. It is not possible to send value along the functions call due to the use of delegatecall.

Reviewer: ```suggestion
Allows a caller to batch different function calls in one call. Performs a `delegatecall` on self to call different functions while preserving the context. It is impossible to send value along the functions call due to the use of delegatecall.
```

I would suggest either using "delegate call" within the text or using code brackets for the function name

> Allows a caller to batch different function calls in one call. Performs a delegatecall on self to call different functions while preserving the context. It is not possible to send value along the functions call due to the use of delegatecall.

Reviewer 2: Suggested rewording

```suggestion
Allows a caller to call multiple functions from the contract interface in one single batch call. 

Under the hood, this function performs a `delegatecall` on the contract itself to call different functions while preserving the context. 

> NB: It is not possible to send value along the functions call due to the use of `delegatecall`.
```

#### Parameters:

| Name   | Type      | Description                                                                                                                     |
| :----- | :-------- | :------------------------------------------------------------------------------------------------------------------------------ |

> | `data` | `bytes[]` | An array of ABI encoded function calls to be called on the contract. The array can contain calls to different functions. |

Reviewer 2: ```suggestion
| `data` | `bytes[]` | An array of ABI encoded function calls to be called on the contract. The array can contain calls to different functions from the contract interface. |
```

_Triggers the **[ContractCreated](#contractcreated)** event on every newly created smart contract that used operation `CREATE` or `CREATE2`._

:::note

> The `execute(uint256[],address[],uint256[],bytes[])` function can only be called by the current owner of the contract and any address allowed by the owner's logic.

Reviewer: ```suggestion
The `execute(uint256[], address[],uint256[], bytes[])` function can only be called by the current owner of the contract and any address allowed by the owner's logic.
```

are there spaces missing on purpose?
